finality-aleph/src/network/manager/discovery.rs

```diff
@@ -348,6 +352,38 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn broadcasts_when_clueless_modulo_3() {
+        let (mut discovery, mut handlers, _) = build_number(3).await;
```

Comment:
could check here also other multiplies of 3 (just 2 or 3 would be enough)

Reply:
Refactored to check more cases.